REGRESSION(267629@main): create_hash_table script runs forever on i686

mcatanzaro opened the pull request:
#### 08395c362d950165a0646c78ce89101979ba2618
<pre>
REGRESSION(267629@main): create_hash_table script runs forever on i686
<a href="https://bugs.webkit.org/show_bug.cgi?id=265554">https://bugs.webkit.org/show_bug.cgi?id=265554</a>

Reviewed by Mark Lam.

This reverts 267629@main, and also adds some additional safety checks to
the create_hash_table script.

bigint and Math::BigInt are two completely different perl modules.
bigint causes the entire script to transparently use Math::BigInt for
mathematical operations. Importing the wrong module effectively sabotaged
the script. Whoops.

bigint appears to be part of a standard perl installation, so for
upstream purposes we can expect it to be always available. However,
Fedora packages the perl interpreter separately from its modules. When I
went to install the bigint module, I tricked myself into thinking it was
already installed because Math::BigInt was installed, and another
developer suggested that the import syntax was wrong, and I improperly
went along with this because I am not very familiar with perl.

So, fix the import syntax to make this script actually use big integers
again. This fixes an infinite loop on 32-bit platforms if perl is not
compiled with 64-bit integer support. It also affects the generated hash
tables, so I had to rebase two of the bindings test results.

Also, add new safety checks to ensure the script fails if the hash
size grows too large. This will prevent the script from infinite looping
if somebody else breaks the script in the future. The guard values I
selected are arbitrary numbers that are much bigger than expected.

* Source/JavaScriptCore/create_hash_table:
* Source/WebCore/bindings/scripts/Hasher.pm:
* Source/WebCore/bindings/scripts/test/JS/JSTestConditionallyReadWrite.cpp:
(WebCore::JSTestConditionallyReadWriteDOMConstructor::prototypeForStructure):
* Source/WebCore/bindings/scripts/test/JS/JSTestGlobalObject.cpp:

Canonical link: <a href="https://commits.webkit.org/271765@main">https://commits.webkit.org/271765@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4c236d780ae1f3a2a8739d44111fe5617c710e0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/29537 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8203 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/30858 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32071 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/26762 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30138 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10355 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5496 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/26766 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/29810 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/6852 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25235 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/5857 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/5990 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26321 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33415 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/25399 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/26956 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/26741 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32212 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/29771 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/5946 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4137 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/29986 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/7680 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/36109 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7020 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/6487 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7783 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6469 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

MenloDorian commented on Source/JavaScriptCore/create_hash_table:
```diff
@@ -148,6 +148,9 @@ sub calcPerfectHashSize($)
     my ($isMac) = @_;
 tableSizeLoop:
     for ($perfectHashSize = ceilingToPowerOf2(scalar @keys); ; $perfectHashSize += $perfectHashSize) {
+        if ($perfectHashSize > 2**15) {
```
Was the `perfectHashSize` rename landed separately?  I no longer see it reflected in this patch.

mcatanzaro replied:
Yes, see #21472

MenloDorian reviewed (approved):
Based on Yijia's feedback, r=me if EWS bots are green.

mcatanzaro committed:
REGRESSION(267629@main): create_hash_table script runs forever on i686
https://bugs.webkit.org/show_bug.cgi?id=265554

Reviewed by Mark Lam.

This reverts 267629@main, and also adds some additional safety checks to
the create_hash_table script.

bigint and Math::BigInt are two completely different perl modules.
bigint causes the entire script to transparently use Math::BigInt for
mathematical operations. Importing the wrong module effectively sabotaged
the script. Whoops.

bigint appears to be part of a standard perl installation, so for
upstream purposes we can expect it to be always available. However,
Fedora packages the perl interpreter separately from its modules. When I
went to install the bigint module, I tricked myself into thinking it was
already installed because Math::BigInt was installed, and another
developer suggested that the import syntax was wrong, and I improperly
went along with this because I am not very familiar with perl.

So, fix the import syntax to make this script actually use big integers
again. This fixes an infinite loop on 32-bit platforms if perl is not
compiled with 64-bit integer support. It also affects the generated hash
tables, so I had to rebase two of the bindings test results.

Also, add new safety checks to ensure the script fails if the hash
size grows too large. This will prevent the script from infinite looping
if somebody else breaks the script in the future. The guard values I
selected are arbitrary numbers that are much bigger than expected.

* Source/JavaScriptCore/create_hash_table:
* Source/WebCore/bindings/scripts/Hasher.pm:
* Source/WebCore/bindings/scripts/test/JS/JSTestConditionallyReadWrite.cpp:
(WebCore::JSTestConditionallyReadWriteDOMConstructor::prototypeForStructure):
* Source/WebCore/bindings/scripts/test/JS/JSTestGlobalObject.cpp:

Canonical link: https://commits.webkit.org/271765@main
